builtins/web/timers.cpp

```diff
@@ -131,9 +131,8 @@ template <bool repeat> bool setTimeout_or_interval(JSContext *cx, const unsigned
   }
 
   if (!(args[0].isObject() && JS::IsCallable(&args[0].toObject()))) {
-    JS_ReportErrorASCII(cx, "First argument to %s must be a function",
-                        repeat ? "setInterval" : "setTimeout");
-    return false;
+    return api::throw_error(cx, api::Errors::WrongType, repeat ? "setInterval" : "setTimeout",
+      "first argument", "be a function");
```

Comment:
Strictly speaking, I'm not sure this is a required error for timers per https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#timers.

Reply:
the reason for this exception is that we don't currently support the invocation mode where a source string is passed in instead of a function. There's no fundamental reason not to do so, but it's not something I want to tackle here.